lib/ball.js

```diff
@@ -22,12 +24,9 @@ Ball.prototype.draw = function () {
 };
 
 Ball.prototype.move = function (bumpers) {
-  // this.bounceCheck(bumpers); ## This may be able to stay here once we refactor the game engine loop
+  this.collisionCheck(bumpers);
   this.x = this.x + (this.xSpeed * this.xDirection);
   this.y = this.y + (this.ySpeed * this.yDirection);
-
-  this.bounceCheck(bumpers);
```

Comment:
This was the bounce check that was put in place to correct the issue with balls going through the wall. I can reproduce the bug with the collision check on line 27.